### ISSUE TITLE
test_backward_node_failure_python_udf is flaky

### ISSUE THREAD

**Issue by mrshenli** [pytorch/pytorch#35099]
Not disabling the test for now as 1) the error is very rare and didn't occur on master yet 2) this might be caused by some race in RPC code instead of the flakiness in this specific test as similar error occurs in #34952 too. 

https://app.circleci.com/pipelines/github/pytorch/pytorch/143234/workflows/1fe63b95-2578-4964-8797-aa107a4758e0/jobs/4886776

```
Mar 20 03:58:32 Running tests...
Mar 20 03:58:32 ----------------------------------------------------------------------
Mar 20 03:58:47 ............s[E request_callback_impl.cpp:94] Received error while processing request type 15: TypeError: 'NoneType' object is not callable
Mar 20 03:59:27 F....................Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:27 Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:27 Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:27 Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:28 .[E request_callback_impl.cpp:94] Received error while processing request type 15: size mismatch, m1: [3 x 3], m2: [6 x 6] at /var/lib/jenkins/workspace/aten/src/TH/generic/THTensorMath.cpp:41
Mar 20 03:59:28 [E request_callback_impl.cpp:94] Received error while processing request type 15: size mismatch, m1: [3 x 3], m2: [6 x 6] at /var/lib/jenkins/workspace/aten/src/TH/generic/THTensorMath.cpp:41
Mar 20 03:59:28 [E request_callback_impl.cpp:94] Received error while processing request type 15: size mismatch, m1: [3 x 3], m2: [6 x 6] at /var/lib/jenkins/workspace/aten/src/TH/generic/THTensorMath.cpp:41
Mar 20 03:59:28 [E request_callback_impl.cpp:94] Received error while processing request type 15: size mismatch, m1: [3 x 3], m2: [6 x 6] at /var/lib/jenkins/workspace/aten/src/TH/generic/THTensorMath.cpp:41
Mar 20 03:59:30 .Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:30 Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:30 Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:30 Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:30 /opt/conda/lib/python3.6/site-packages/torch/nn/functional.py:1816: UserWarning: Argument order of nn.functional.embedding_bag was changed. Usage `embedding_bag(weight, input, ...)` is deprecated, and should now be `embedding_bag(input, weight, ...)`.
Mar 20 03:59:30   warnings.warn("Argument order of nn.functional.embedding_bag was changed. "
Mar 20 03:59:30 /opt/conda/lib/python3.6/site-packages/torch/nn/functional.py:1816: UserWarning: Argument order of nn.functional.embedding_bag was changed. Usage `embedding_bag(weight, input, ...)` is deprecated, and should now be `embedding_bag(input, weight, ...)`.
Mar 20 03:59:30   warnings.warn("Argument order of nn.functional.embedding_bag was changed. "
Mar 20 03:59:30 /opt/conda/lib/python3.6/site-packages/torch/nn/functional.py:1816: UserWarning: Argument order of nn.functional.embedding_bag was changed. Usage `embedding_bag(weight, input, ...)` is deprecated, and should now be `embedding_bag(input, weight, ...)`.
Mar 20 03:59:30   warnings.warn("Argument order of nn.functional.embedding_bag was changed. "
Mar 20 03:59:30 /opt/conda/lib/python3.6/site-packages/torch/nn/functional.py:1816: UserWarning: Argument order of nn.functional.embedding_bag was changed. Usage `embedding_bag(weight, input, ...)` is deprecated, and should now be `embedding_bag(input, weight, ...)`.
Mar 20 03:59:30   warnings.warn("Argument order of nn.functional.embedding_bag was changed. "
Mar 20 03:59:53 .................Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:53 Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:53 Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:53 Warning: Using sparse tensors in TorchScript is experimental. Many optimization pathways have not been thoroughly tested with sparse tensors. Please include the fact that the network is running sparse tensors in any bug reports submitted. (operator() at /var/lib/jenkins/workspace/torch/csrc/jit/python/pybind_utils.h:684)
Mar 20 03:59:53 /opt/conda/lib/python3.6/site-packages/torch/nn/functional.py:1816: UserWarning: Argument order of nn.functional.embedding_bag was changed. Usage `embedding_bag(weight, input, ...)` is deprecated, and should now be `embedding_bag(input, weight, ...)`.
Mar 20 03:59:53   warnings.warn("Argument order of nn.functional.embedding_bag was changed. "
Mar 20 03:59:53 /opt/conda/lib/python3.6/site-packages/torch/nn/functional.py:1816: UserWarning: Argument order of nn.functional.embedding_bag was changed. Usage `embedding_bag(weight, input, ...)` is deprecated, and should now be `embedding_bag(input, weight, ...)`.
Mar 20 03:59:53   warnings.warn("Argument order of nn.functional.embedding_bag was changed. "
Mar 20 03:59:53 /opt/conda/lib/python3.6/site-packages/torch/nn/functional.py:1816: UserWarning: Argument order of nn.functional.embedding_bag was changed. Usage `embedding_bag(weight, input, ...)` is deprecated, and should now be `embedding_bag(input, weight, ...)`.
Mar 20 03:59:53   warnings.warn("Argument order of nn.functional.embedding_bag was changed. "
Mar 20 03:59:53 /opt/conda/lib/python3.6/site-packages/torch/nn/functional.py:1816: UserWarning: Argument order of nn.functional.embedding_bag was changed. Usage `embedding_bag(weight, input, ...)` is deprecated, and should now be `embedding_bag(input, weight, ...)`.
Mar 20 03:59:53   warnings.warn("Argument order of nn.functional.embedding_bag was changed. "
Mar 20 04:00:01 ........
Mar 20 04:00:01 ======================================================================
Mar 20 04:00:01 FAIL [6.131s]: test_backward_node_failure_python_udf (__main__.DistAutogradTestWithSpawn)
Mar 20 04:00:01 ----------------------------------------------------------------------
Mar 20 04:00:01 Traceback (most recent call last):
Mar 20 04:00:01   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 175, in wrapper
Mar 20 04:00:01     self._join_processes(fn)
Mar 20 04:00:01   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 285, in _join_processes
Mar 20 04:00:01     self._check_return_codes(elapsed_time)
Mar 20 04:00:01   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 324, in _check_return_codes
Mar 20 04:00:01     self.assertEqual(p.exitcode, first_process.exitcode)
Mar 20 04:00:01   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 915, in assertEqual
Mar 20 04:00:01     super(TestCase, self).assertLessEqual(abs(x - y), prec, message)
Mar 20 04:00:01 AssertionError: 11 not less than or equal to 1e-05 : 
Mar 20 04:00:01 
Mar 20 04:00:01 ----------------------------------------------------------------------
```

cc @ezyang @gchanan @zou3519 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528 @jjlilley @osalpekar

**Comment by mrshenli** [comment]
occurred on master: https://app.circleci.com/pipelines/github/pytorch/pytorch/149107/workflows/e41a5d0d-618c-436e-9517-a1ffe1cd7cfe/jobs/5009799

**Comment by mrshenli** [comment]
fixed by #35394 and  #35263
